src/freenas/usr/local/sbin/enc_helper

```diff
@@ -56,116 +48,53 @@ if __name__ == "__main__":
             else:
-                print("Not detaching GELI providers because an encrypted zpool with name '%s' is still mounted!" % vol.vol_name)
-                log.warn("Not detaching GELI providers because an encrypted zpool with name '%s' is still mounted!" % vol.vol_name)
+                print("Escrowd is running without the passphrase")
+                log.warn("Escrowd is running without the passphrase")
+        elif cmd == 'attachall':
+            c.call('failover.encryption_attachall', timeout=600, job=True, callback=jb_callback)
+
+        elif cmd == 'detachall':
+            job = c.call('failover.encryption_detachall', job='RETURN', callback=jb_callback)
+            if job.result() is False:
                 sys.exit(1)
-    
-    elif cmd == 'interactive':
-        passphrase = getpass.getpass()
-        if passphrase:
-            rv = escrowctl.setkey(passphrase)
-            try:
-                with client as c:
-                    c.call('failover.call_remote', 'failover.encryption_setkey', [passphrase])
-            except:
-                print("Failed to set the passphrase on the standby node.")
-                log.warn("Failed to set the passphrase on the standby node.")
-            if not args.key_only:
-                notifier().failover_force_master()
-    else:
-        if cmd == 'synctopeer':
-            passphrase = escrowctl.getkey()
+
+        elif cmd == 'interactive':
+            passphrase = getpass.getpass()
             if passphrase:
-                with client as c:
+                rv = c.call('failover.encryption_setkey', passphrase)
+                try:
                     c.call('failover.call_remote', 'failover.encryption_setkey', [passphrase])
-            else:
-                print("ERROR: passphrase not detected.")
-                log.warn("ERROR: passphrase not detected.")
-        elif cmd == 'syncfrompeer':
-            with client as c:
+                except Exception:
+                    print("Failed to set the passphrase on the standby node.")
+                    log.warn("Failed to set the passphrase on the standby node.")
+                if not args.key_only:
+                    c.call('failover.force_master')
+        else:
+            if cmd == 'synctopeer':
+                passphrase = c.call('failover.encryption_getkey')
+                if passphrase:
+                    c.call('failover.call_remote', 'failover.encryption_setkey', [passphrase])
+                else:
+                    print("ERROR: passphrase not detected.")
+                    log.warn("ERROR: passphrase not detected.")
+            elif cmd == 'syncfrompeer':
                 passphrase = c.call('failover.call_remote', 'failover.encryption_getkey')
-            rv = escrowctl.setkey(passphrase)
-        if not rv:
-            print("Error")
+                rv = c.call('failover.encryption_setkey', passphrase)
+            if not rv:
+                print("Error")
```

Comment:
Can we print a reasonable error here and state something like "ERROR: failed to set passphrase on this controller" or something like that?

Reply:
Sure, updated